tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -20,16 +22,33 @@ internal class LambdaCommon
         private const string DefaultJson = "{}";
         private const double ServerlessMaxWaitingFlushTime = 3;
 
-        internal static CallTargetState StartInvocation<TArg>(TArg payload, ILambdaExtensionRequest requestBuilder)
+        internal static CallTargetState StartInvocation<TArg>(TArg payload, ILambdaExtensionRequest requestBuilder, IDictionary<string, string> context)
         {
             var json = SerializeObject(payload);
-            NotifyExtensionStart(requestBuilder, json);
+            NotifyExtensionStart(requestBuilder, json, context ?? new Dictionary<string, string>());
             return new CallTargetState(CreatePlaceholderScope(Tracer.Instance, requestBuilder));
         }
 
         internal static CallTargetState StartInvocationWithoutEvent(ILambdaExtensionRequest requestBuilder)
         {
-            return StartInvocation(DefaultJson, requestBuilder);
+            return StartInvocation(DefaultJson, requestBuilder, null);
+        }
+
+        internal static CallTargetState StartInvocationOneParameter<TArg>(TArg eventOrContext, ILambdaExtensionRequest requestBuilder)
+        {
+            if (eventOrContext.GetType().ToString().Contains("LambdaContext"))
+            {
+                var dict = GetTraceContext(eventOrContext);
+                return StartInvocation(DefaultJson, requestBuilder, dict as IDictionary<string, string>);
+            }
+
+            return StartInvocation(eventOrContext, requestBuilder, null);
+        }
+
+        internal static CallTargetState StartInvocationTwoParameters<TArg1, TArg2>(TArg1 payload, ILambdaExtensionRequest requestBuilder, TArg2 context)
+        {
+            var dict = GetTraceContext(context);
+            return StartInvocation(payload, requestBuilder, dict as IDictionary<string, string>);
```

Comment:
If `GetTraceContext()` returns `IDictionary<string, string>` you can remove this cast.

Reply:
removed